src/main/java/uk/gov/hmcts/dm/config/batch/BatchConfiguration.java

```diff
@@ -108,11 +112,28 @@ public Step step1() {
             .reader(undeletedDocumentsWithTtl())
             .processor(deleteExpiredDocumentsProcessor)
             .writer(itemWriter())
-            .taskExecutor(new SimpleAsyncTaskExecutor("del_w_ttl"))
+            .taskExecutor(taskExecutor())
             .build();
 
     }
 
+    private ThreadPoolTaskExecutor taskExecutor() {
+        ThreadPoolTaskExecutor taskExecutor = new ThreadPoolTaskExecutor();
+        taskExecutor.setThreadNamePrefix("del_w_ttl");
+        taskExecutor.setCorePoolSize(deleteThreadCount);
+        taskExecutor.setMaxPoolSize(deleteThreadCount);
+        taskExecutor.setQueueCapacity(10);
```

Comment:
Can we parameterise the queue capacity as well?

Reply:
removed